challenge-sizebay/src/Components/ProgressBar/ProgressBar.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useContext } from "react";
+import { useLayoutEffect } from "react";
+import { GlobalContext } from "../../Contexts/GlobalContext";
+import { ProgressWrapper, ProgressDone } from "./style";
+import confetti from 'canvas-confetti'
+import { useEffect } from "react";
+
+const ProgressBar = () => {
+  const [style, setStyle] = useState({});
+  const { states } = useContext(GlobalContext);
+
+  const percentageDone = (part, total) => part * 100 / total;
+
+  useLayoutEffect(() => {
+    const newStyle = {
+      width: `${percentageDone(states.done.length, states.pending.length + states.done.length)}%`,
+    };
+    setStyle(newStyle);
+
+  }, [states.pending.length, states.done.length]);
```

Comment:
@luizfbarbosa12 just curious: why have you decided to go for a `useLayoutEffect` instead of the usual `useEffect`?

Reply:
@luizfbarbosa12 that's exactly what I thought, just wanted to have you covering it since it's kind of an obscure-ish hook. Nice implementation!